### PR TITLE
save last summarized op number to database

### PR DESCRIPTION
**Pull request by appym31**
Save the protocol head (i.e. last summarized sequence number) to the database so that it can be read later from scriptorium to delete summarized ops in the database. Update IDocument so that the lastSummarizedSequenceNumber can be read from it.

### CODE REVIEW

File: server/routerlicious/packages/test-utils/src/testDocumentStorage.ts [hunk 1]
```diff
@@ -191,6 +192,7 @@ export class TestDocumentStorage implements IDocumentStorage {
                 scribe: undefined,
                 tenantId,
                 version: "0.1",
+                lastSummarizedSequenceNumber: -1,
```
Comment (tanviraumi) [inline server/routerlicious/packages/test-utils/src/testDocumentStorage.ts:195]: This is a scribe property so this should be inside scribe object. Not a top level one.

File: server/routerlicious/packages/lambdas/src/scribe/lambda.ts [hunk 2]
```diff
@@ -311,6 +311,7 @@ export class ScribeLambda implements IPartitionLambda {
                                     operation.sequenceNumber,
                                     false,
                                     this.serviceConfiguration.scribe.clearCacheAfterServiceSummary);
+                                this.updateProtocolHead(operation.sequenceNumber);
```
Comment (tanviraumi) [inline server/routerlicious/packages/lambdas/src/scribe/lambda.ts:314]: As a start, let's delete the ops that client has summarized. Don't update the protocolhead after a service summary.

File: server/routerlicious/packages/services-core/src/document.ts [hunk 3]
```diff
@@ -112,4 +112,6 @@ export interface IDocument {
 
     // Deli state
     deli: string;
+
+    lastSummarizedSequenceNumber: number;
```
Comment (pradeepvairamani) [inline server/routerlicious/packages/services-core/src/document.ts:116]: Shouldn't this be in IScribe?

File: server/routerlicious/packages/test-utils/src/testDocumentStorage.ts [hunk 4]
```diff
@@ -121,6 +121,7 @@ export class TestDocumentStorage implements IDocumentStorage {
             },
             sequenceNumber,
             lastClientSummaryHead: undefined,
+            lastSummarySequenceNumber: -1,
```
Comment (tanviraumi) [inline server/routerlicious/packages/test-utils/src/testDocumentStorage.ts:124]: We always start from sequence number 1. So the default value should be 0 everywhere.

File: server/routerlicious/packages/services-core/src/document.ts [hunk 5]
```diff
@@ -94,6 +94,9 @@ export interface IScribe {
 
     // Ref of the last client generated summary
     lastClientSummaryHead: string | undefined;
+
+    // Sequence number of the last summarized op
```
Comment (tanviraumi) [inline server/routerlicious/packages/services-core/src/document.ts:98]: Sequence number of last operation that was part of latest summary.

File: server/routerlicious/packages/lambdas/src/scribe/lambda.ts [hunk 6]
```diff
@@ -514,6 +521,15 @@ export class ScribeLambda implements IPartitionLambda {
         this.protocolHead = protocolHead;
     }
 
+    /**
+     * lastSummarySequenceNumber is the sequence number of the last summary
+     * This method updates it to the latest summary sequence number
+     * @param summarySequenceNumber The sequence number of the latest summary
+     */
```
Comment (tanviraumi) [inline server/routerlicious/packages/lambdas/src/scribe/lambda.ts:528]: Tracks the last sequence number that was part of latest summary

**Review by tanviraumi — APPROVED** [review]
Checkout the comments. Otherwise looks good.

**Pull request merged**